src/foam/util/AddressUtil.java

```diff
@@ -69,48 +93,81 @@ public static String[] parseAddress(String address1, String address2) throws Ill
     return new String[] { streetNumber, streetName };
   }
 
-  public static String normalizeRegion(X x, String country, String regionCode) {
-    if ( SafetyUtil.isEmpty(regionCode) || SafetyUtil.isEmpty(country) ) {
-      return regionCode;
+  public static String normalizeRegion(X x, String country, String region) {
+    if ( SafetyUtil.isEmpty(region) || SafetyUtil.isEmpty(country) ) {
+      return region;
+    }
+
+    country = country.trim();
+    region = region.trim();
+
+    Region regionObject = lookupRegion(x, country, region);
+    if ( regionObject != null ) {
+      return regionObject.getCode();
+    } else {
+      return country + "-" + region;
     }
+  }
+
+  public static String normalizeRegion(X x, String country, String region, String postalCode) {
+    if ( SafetyUtil.isEmpty(region) || SafetyUtil.isEmpty(country) ) {
```

Comment:
we can still satisfy region lookup if country is empty via post code lookup, or default to CA